### ISSUE TITLE
Don't enforce new versions

### ISSUE THREAD

**Issue by sidneycrestani** [crmarsh/discord-linux-bugs#121]
The last update(0.0.9 to 0.0.10) should've had an option like 'stay on current version', so if the new version breaks stuff we can downgrade and keep using the app.

**Comment by crmarsh** [comment]
I'm still thinking about it but here's where I'm at right now:
* the canary build is all about having people testing the latest build; I don't think I'd change it there.
* Perhaps for the stable build there should be a grace period.
* I'd prefer putting work into stuff that makes updating easier.

**Comment by anders-larsson** [comment]
This is especially frustrating on distributions where the Discord is not packaged by the Discord team (and more exactly you I suppose). The situation on Linux distributions is that you almost always want to use your distributions package manager to maintain packages. Since the Discord team only packages the application (which I totally understand) for the Debian family right now this makes the application unusable for users of other distributions until the new version of the application has been released that specific distribution. They could of course download the archive themselves and unpack and run it. But it pretty much defeats the purpose distributing packages on Linux.

This is at least my point of view of this issue.

**Comment by crmarsh** [comment]
Yeah, I get it. I think (per my third point) I'd rather just expand how many distros we automatically update things for -- I mean, I don't package up a deb build by hand, I have a script on a build machine, I don't think it should be a big problem adding some more outputs to that script (particularly if I can leverage the existing scripts people have already made to do this for themselves).

**Comment by maxpoulin64** [comment]
Adding to this, in the Linux world there are many reasons why a user could possibly not be able to update. When it's installed properly as a package, the user would need root access to be able to install the update. I understand people are sort of expected to have root access to their machines because gaming and all, but if the computer is centrally managed (kids, family, schools, workplace break room, etc.) or simply not yours (guests), they have to wait for me (or someone that has root access and knows how to package an update) to be available to package and deploy the update. It's extremely annoying.

If anything, a `--no-updates-i-know-what-im-doing` launch flag or something would be really nice to have.

(Software in general shouldn't auto-update itself as it's the distro's package manager's job to do that, but that I can pass as most people uses Windows and Windows users are the worst at keeping anything updated if they're not forced into it.)

**Comment by 2E0PGS** [comment]
I currently have version 0.0.2 installed on Ubuntu and voice chat is broken. I could just roll back but I cannot since updates are forced so if I install 0.0.1 again I get forced to update to new version or the program will not open. Adding launch flags for this would also be very useful. Besides that a simple do not update option is needed.

**Comment by Serkan-devel** [comment]
This could be somewhat solved here https://github.com/crmarsh/discord-linux-bugs/issues/205

**Comment by 2E0PGS** [comment]
@Serkan-devel nope not if the app image or other package still have forced updates! I have the old deb file and can install it but Discord wont open unless I update to the 0.0.2 deb. Same would happen if I used an app image or docker image or whatever package system.

**Comment by Serkan-devel** [comment]
Appimage's update feature is separate but @probonopd might know more.

**Comment by probonopd** [comment]
@Serkan-devel what do you want to know?

**Comment by Serkan-devel** [comment]
If updates can be suppressed on AppimageUpdate

**Comment by 2E0PGS** [comment]
@Serkan-devel It's nothing to do with the package! It is to do with Discords code. They are using their own updating system. So it has nothing to do with system updates via PPA's or manual appimage updates etc. All they need to do is add an option to stay on current version during version check popup for host application updates.

**Comment by Serkan-devel** [comment]
oh, now I get it

**Comment by probonopd** [comment]
I don't understand the question @Serkan-devel. AppimageUpdate checks for updates if it is invoked by the user. If you don't want to run AppimageUpdate, then simply don't run it.

**Comment by Serkan-devel** [comment]
nvm
